### PR TITLE
Get logs by hash for blocks close to tip

### PR DESCRIPTION
**Pull request by sz-piotr**

### CODE REVIEW

File: packages/backend/src/peripherals/ethereum/EthereumClient.ts [hunk 1]
```diff
@@ -36,11 +36,33 @@ export class EthereumClient {
     if (blockRange.isEmpty()) {
       return []
     }
-    const logs = await this.provider.getLogs({
-      ...filter,
-      fromBlock: blockRange.start,
-      toBlock: blockRange.end - 1,
-    })
+    let [from, to, hashes] = blockRange.splitByKnownHashes()
+    if (hashes.length > 40) {
```
Comment (krzkaczor) [inline packages/backend/src/peripherals/ethereum/EthereumClient.ts:40]: should 40 be SAFE_BLOCK_DISTANCE ?

File: packages/backend/src/peripherals/ethereum/EthereumClient.ts [hunk 2]
```diff
@@ -36,11 +36,33 @@ export class EthereumClient {
...
+
+    const nestedLogs = await Promise.all([
+      from !== to
+        ? this.provider.getLogs({
+            ...filter,
+            fromBlock: from,
+            toBlock: to - 1,
+          })
+        : [],
+      ...hashes.map((blockHash) =>
```
Comment (krzkaczor) [inline packages/backend/src/peripherals/ethereum/EthereumClient.ts:58]: Have you tested it in practice? I think we might need to throttle concurrent calls here 🤔

Comment (sz-piotr) [reply packages/backend/src/peripherals/ethereum/EthereumClient.ts]: I have but I implemented batching anyway

File: packages/backend/src/peripherals/ethereum/EthereumClient.ts (hunk 1 above)
Comment (sz-piotr) [reply packages/backend/src/peripherals/ethereum/EthereumClient.ts]: Moved the value to config

**Pull request merged**